lib/chef/provisioning/oneview/oneview_api.rb

```diff
@@ -374,40 +374,96 @@ def customize_machine(action_handler, machine_spec, machine_options, profile)
+            'domainType' => machine_options[:driver_options][:domainType],
+            'domainName' => machine_options[:driver_options][:domainName],
+            'nics'       => nics
+          }
+        }] }
+        network_personalization_task = rest_api(:icsp, :put, '/rest/os-deployment-apxs/personalizeserver', options)
+        network_personalization_task_uri = network_personalization_task['uri']
+        60.times do # Wait for up to 10 min
+          network_personalization_task = rest_api(:icsp, :get, network_personalization_task_uri, options)
+          break if network_personalization_task['running'] == 'false'
+          print "."
+          sleep 10
+        end
+        unless network_personalization_task['state'] == 'STATUS_SUCCESS'
+          raise "Error performing network personalization: #{network_personalization_task['jobResult'].first['jobResultLogDetails']}\n#{network_personalization_task['jobResult'].first['jobResultErrorDetails']}"
+        end
+        #Add check to see if task reports success. Bail if not success. Check ICSP IP configuration to see if it matches the machine options IP. Do in loop for "10 minutes" 
+        requested_ips = []
+        machine_options[:driver_options][:connections].each do |connection|
+          if connection[1][:dhcp] == false
+            requested_ips.push connection[1][:ip4Address]
+          end
+        end
+        60.times do 
+          my_server_connections = rest_api(:icsp, :get, my_server['uri'])["interfaces"]
+          my_server_connections.each do |connection|
+            if requested_ips.include? connection["ipv4Addr"]
+              requested_ips.delete connection["ipv4Addr"]
+            end
+          end  
+
+          break if requested_ips.empty?  
+          print "."
+          sleep 10    
+        end
+
+        if !requested_ips.empty? 
+          raise "Error setting up ips correctly in ICSP"
+        end
+        action_handler.perform_action "Perform network flipping on #{machine_spec.name}" do
+          action_handler.report_progress "INFO: Performing network flipping on #{machine_spec.name}"
+          if machine_options[:driver_options][:connections]
+            machine_options[:driver_options][:connections].each do |id, data|
+              if data[:net] && data[:deployNet]
+                available_networks = rest_api(:oneview, :get, "/rest/server-profiles/available-networks?serverHardwareTypeUri=#{profile['serverHardwareTypeUri']}&enclosureGroupUri=#{profile['enclosureGroupUri']}")
+                deploy_network = nil
+                new_network = nil
+                available_networks['ethernetNetworks'].each do |network|
+                  if network['name'] == data[:net]
+                    new_network = network
+                  elsif network['name'] == data[:deployNet]
+                    deploy_network = network
+                  end
+                end
+                if new_network && deploy_network
```

Comment:
@mbfrahry , do we want to print any warning or throw an error if the networks don't exist?